pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,153 +1,226 @@
 {
+  lib,
+  stdenv,
   bzip2,
   callPackage,
-  expat,
   fetchFromGitHub,
   fontconfig,
   freetype,
   glib,
   glslang,
   harfbuzz,
-  lib,
-  libadwaita,
   libGL,
-  libpng,
   libX11,
-  libXcursor,
-  libXi,
-  libXrandr,
+  libadwaita,
   ncurses,
   nixosTests,
   oniguruma,
   pandoc,
   pkg-config,
   removeReferencesTo,
-  stdenv,
   versionCheckHook,
   wrapGAppsHook4,
   zig_0_13,
-  zlib,
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/apprt.zig#L72-L76
+  appRuntime ?
+    {
+      linux = "gtk";
+      windows = "glfw";
+    }
+    .${stdenv.hostPlatform.parsed.kernel.name} or "none",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/font/main.zig#L94
+  fontBackend ? if stdenv.hostPlatform.isDarwin then "coretext" else "fontconfig_freetype",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/PACKAGING.md#build-options
+  optimizeLevel ? "ReleaseFast",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/renderer.zig#L51-L52
+  renderer ? if stdenv.hostPlatform.isDarwin then "metal" else "opengl",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/build.zig#L106
+  withAdwaita ? true,
 }:
+
+# https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/apprt.zig#L59-L83
+assert lib.assertOneOf "appRuntime" appRuntime [
+  "glfw"
+  "gtk"
+  "none"
+];
+
+# https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/font/main.zig#L51-L76
+# TODO: Support `web_canvas` for WASM?
+assert lib.assertOneOf "fontBackend" fontBackend [
+  "coretext"
+  "coretext_freetype"
+  "coretext_harfbuzz"
+  "coretext_noshape"
+  "fontconfig_freetype"
+  "freetype"
```

Comment:
Is this something you'd realistically want to customise?

Reply:
I'd imagine some people do given how many options there are. It seems Mac-centric though with how many coretext options there are; so it'll hopefully be more useful when we unbreak Darwin

Reply:
In that case I'd prefer we only added this when darwin support is added and people actually care to configure this.

Reply:
Even when Darwin support is added, Ghostty will automatically detect the correct backends to use. Allowing the user to customize this will only result in broken packages and a lot of bug reports. Any choice that doesn't match what Ghostty autodetects is going to be broken. None of them are tested by Ghostty and are only occasionally used during development.